spec/requests/user_can_create_group_spec.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+describe 'POST /groups' do
+  let(:user) { create(:user) }
+
+  describe 'POST req with valid credentials' do
+    let(:user_credentials) { user.create_new_auth_token }
+    let(:headers) { { HTTP_ACCEPT: 'application/json' }.merge!(user_credentials) }
+    
+    before do
+      post "/groups", params: { group: {name: 'coding'} }, headers: headers
+    end
+
+    it 'responds with success message' do
+      expect(response_json['message']).to eq 'You have created a group successfully'
+    end
+
+    it 'responds with status 200' do
+      expect(response).to have_http_status 200
+    end
+  end
+end
```

Comment:
You need to add a spec for failing to create a group

Reply:
Of course.  Has been addressed and fixed.